microsoft/testsuites/kselftest/kselftest.py

```diff
@@ -99,27 +103,31 @@ def __init__(
         # tar file path specified in yml
         self._tar_file_path = kselftest_file_path
         if self._tar_file_path:
+            self._remote_tar_path = (
+                self.node.get_pure_path("/tmp/kselftest") / os.path.basename(self._tar_file_path)
+            )
+            self._kself_installed_dir = (
+                self.node.get_pure_path("/tmp/kselftest") / "kselftest-packages"
+            )
+        else:
             self._remote_tar_path = self.get_tool_path(
                 use_global=True
             ) / os.path.basename(self._tar_file_path)
-
-        # command to run kselftests
-        self._kself_installed_dir = (
-            self.get_tool_path(use_global=True) / "kselftest-packages"
-        )
+            self._kself_installed_dir = (
+                self.get_tool_path(use_global=True) / "kselftest-packages"
+            )
 
         self._command = self._kself_installed_dir / "run_kselftest.sh"
 
     # install common dependencies
     def _install(self) -> bool:
-        if not (
-            (
-                isinstance(self.node.os, Ubuntu)
-                and self.node.os.information.version >= "18.4.0"
-            )
-            or isinstance(self.node.os, CBLMariner)
+        if (isinstance(self.node.os,
```

Comment:
You don't need to list all of them here. 1) The BSD covers FreeBSD and OpenBSD, etc, 2) Some uncommon distros doesn't list here, like NixOS, OtherLinux.

Reply:
Please check relationships of all distros in the operating_system.py, for example, the Fedora includes Redhat, CentOS.